include/configs/rpi.h

```diff
@@ -31,4 +31,9 @@
  */
 #define CFG_SYS_SDRAM_SIZE		SZ_128M
 
+/* Environment */
+ 
+#define CONFIG_BOOTCOUNT_ENV
```

Comment:
(Mindprobe: in my mind these setting are doing the same thing as modifying the .config, they are just set in the header file.)

You're already modifying the defconfigs.
Why have this separately set and in a header file?

Or if the reason was to not repeat the same config for every defconfig, why not translate
```
CONFIG_ENV_IS_IN_MMC=y
CONFIG_ENV_OFFSET=0x400000
CONFIG_ENV_OFFSET_REDUND=0x1000000
CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
```

here and remove all the defconfig patches?

Reply:
@TheMeaningfulEngineer after thinking a bit about it, actually I think the header modifications should be removed. It seems to be historic clutter for unknown reasons, but I'd definitely say that those settings should go to the respective `defconfigs`, just like they do on Yocto already. Hence moved them there to keep things in line.